### PR TITLE
Add new Object Inspector UI to Source viewer (as experiment)

### PR DESCRIPTION
**Pull request by bvaughn**
Loom walk through:
https://www.loom.com/share/f430d276aeb04263afc82b98770eefdf

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,6 +50,7 @@
     "@tiptap/react": "^2.0.0-beta.108",
     "@tiptap/starter-kit": "^2.0.0-beta.183",
     "base64-arraybuffer": "^1.0.2",
+    "bvaughn-architecture-demo": "file:packages/bvaughn-architecture-demo",
```
Comment (bvaughn) [inline package.json:53]: This is kind of gross, but I'm not sure if I want to go through making a new Workspace package _yet_, at least until Mark's done deciding what our new package manager will be.

File: src/devtools/client/debugger/src/components/Editor/Preview/NewObjectInspector.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  // At the moment this means that we need to convert the ValueFront back to a protocol Value.
+  const protocolValue: ProtocolValue | null = useMemo(() => {
+    if (pause == null || pause.pauseId == null) {
+      return null;
+    } else if (preview == null || !preview.hasOwnProperty("root") == null) {
+      return null;
+    }
+
+    // TODO Move this into an action creator or whatever loads the Pause data.
+    preCacheObjects(pause.pauseId, Array.from(pause.objects.values()));
```
Comment (bvaughn) [inline src/devtools/client/debugger/src/components/Editor/Preview/NewObjectInspector.tsx:29]: I'm going to move this to a Thunk or something before I mark the PR as ready for review.

Comment (bvaughn) [reply src/devtools/client/debugger/src/components/Editor/Preview/NewObjectInspector.tsx:29]: I might be able to pass these objects through the `clientValueToProtocolValue` before passing them to `preCacheObjects` – which would let me get rid of the hack in `HTMLChildrenRenderer`.

**Review by jasonLaster — COMMENTED** [review]
Both excited to see the new OI in our app and sad to see some of your beautiful code compromised by the realities of our app 😄

**Comment by bvaughn** [comment]
Okay. I think this is ready for review now.

**Review by markerikson — APPROVED** [review]

**Pull request merged**